### PR TITLE
use kwarg arch from ipex

### PR DESCRIPTION
**Pull request by guangyey**
# Motivate
enable triton.compile even in multi-processing.

### CODE REVIEW

File: python/__init__.py [hunk 1]
```diff
@@ -453,6 +453,10 @@ def get_stream(self, idx=None):
     def get_device_properties(self, device):
         return self.driver.utils.get_device_properties(torch.xpu.device(device).sycl_device)  # noqa: E501
 
+    @functools.lru_cache(None)
+    def get_device_capability(self, device):
```
Comment (chengjunlu) [inline python/__init__.py:457]: Use the get_device_properties instead.

File: python/__init__.py [hunk 2]
```diff
@@ -470,10 +474,12 @@ def get_kernel_bin(self):
         return "spvbin"
 
     def get_architecture_descriptor(self, **kwargs):
-        dev_props = self.get_device_properties(self.get_current_device())
-        max_work_group_size = dev_props['max_work_group_size']
-        max_num_sub_groups = dev_props['max_num_sub_groups']
-        sub_group_sizes = dev_props['sub_group_sizes']
+        cc = kwargs.get("cc", None)
```
Comment (chengjunlu) [inline python/__init__.py:477]: Use the keyword "arch" to align the Triton code. The "cc" is dedicate for CUDA capability

Comment (guangyey) [reply python/__init__.py:477]: Can we also keep compatible with kwarg cc. In pytorch, its code uses cc only, we want to reuse these code.

Comment (guangyey) [reply python/__init__.py:477]: we have an abstraction named get_compute_capability in pytorch as a input to kwarg cc, like https://github.com/pytorch/pytorch/blob/ac60638c6c74149f05c7980f571afa6fe0661b0c/torch/_dynamo/device_interface.py#L142

**Review by chengjunlu — APPROVED** [review]

File: python/__init__.py [hunk 3]
```diff
@@ -470,10 +470,12 @@ def get_kernel_bin(self):
...
-        dev_props = self.get_device_properties(self.get_current_device())
-        max_work_group_size = dev_props['max_work_group_size']
-        max_num_sub_groups = dev_props['max_num_sub_groups']
-        sub_group_sizes = dev_props['sub_group_sizes']
+        arch = kwargs.get("arch", None)
+        if arch is None:
+            arch = self.get_device_properties(self.get_current_device())
+        max_work_group_size = arch['max_work_group_size']
+        max_num_sub_groups = arch['max_num_sub_groups']
+        sub_group_sizes = arch['sub_group_sizes']
```
Comment (EikanWang) [inline python/__init__.py:478]: What is the behavior if the value is None?

Comment (guangyey) [reply python/__init__.py:478]: raise a error, it is expected.

**Pull request merged**